pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -965,7 +965,9 @@ def test_dt64arr_sub_dt64object_array(self, box_with_array, tz_naive_fixture):
         obj = tm.box_expected(dti, box_with_array)
         expected = tm.box_expected(expected, box_with_array)
 
-        warn = PerformanceWarning if box_with_array is not pd.DataFrame else None
+        warn = None
+        if box_with_array is not pd.DataFrame or tz_naive_fixture is None:
+            warn = PerformanceWarning
```

Comment:
Do you know why this changed? We do / do not raise a warning on the array-level?

Reply:
We don't do a warning when operating `op(arr, length_1_object_array)`, which turns out to be the cases described here (in reverse)